src/components/ConnectWallet/index.tsx

```diff
@@ -1,25 +1,26 @@
-import { connections } from 'hooks/connectWeb3/useConnect'
+import useActiveWeb3React, { connections } from 'hooks/connectWeb3/useActiveWeb3React'
 import { useEffect } from 'react'
 
 import ConnectedWalletChip from './ConnectedWalletChip'
 import ConnectWallet from './ConnectWallet'
 
 interface WalletProps {
   disabled?: boolean
-  account?: string
-  onConnectWallet?: (e?: React.MouseEvent<HTMLButtonElement>) => void
+  onClickConnectWallet?: (e?: React.MouseEvent<HTMLButtonElement>) => void
 }
 
-export default function Wallet({ disabled, account, onConnectWallet }: WalletProps) {
+export default function Wallet({ disabled, onClickConnectWallet }: WalletProps) {
   // Attempt to connect eagerly on mount
   useEffect(() => {
-    connections.forEach(([wallet, _]) => wallet.connectEagerly())
+    connections.forEach(([wallet, _]) => wallet?.connectEagerly?.())
```

Comment:
Can you add a comment here that this will immediately trigger lazy loading of the wallets.

Reply:
@zzmp is that different from the `// Attempt to connect eagerly on mount` comment?

@vm if the user isn't already connected to the page, I don't think we want to call activate unless the user clicks `Connect wallet`